GamepadButtonAndAxisEvents/explainer.md

```diff
@@ -0,0 +1,201 @@
+
+```js
+// Example gamepadchange Event:
+gamepadchange {
+  type: "gamepadchange",
+  gamepadSnapshot: Gamepad {id: "Example gamepad", index: 0, …},
+  axesChanged: [0, 1, 2, 3],
+  buttonsChanged: [0],
+  buttonsPressed: [0],
+  buttonsReleased: [],
+}
+
+rawgamepadchange {
+    type: "rawgamepadchange",
+  gamepadSnapshot: Gamepad {id: "Example gamepad", index: 0, …},
+  axesChanged: [0, 1, 2, 3],
+  buttonsChanged: [0],
+  buttonsPressed: [0],
+  buttonsReleased: [],
+}
+```
+```
+// Web IDL for the gamepadchange and rawgamepadchange event.
+
+interface GamepadChangeEvent : Event {
+  const DOMString type = "gamepadchange";
+  readonly attribute Gamepad gamepadSnapshot;
+  readonly attribute FrozenArray<long> axesChanged;
+  readonly attribute FrozenArray<long> buttonsChanged;
+  readonly attribute FrozenArray<long> buttonsPressed;
+  readonly attribute FrozenArray<long> buttonsReleased;
+
+  sequence<GamepadChangeEvent> getCoalescedEvents();
+};
+
+interface RawGamepadChangeEvent : Event {
+  const DOMString type = "rawgamepadchange";
+  readonly attribute Gamepad gamepadSnapshot;
+  readonly attribute FrozenArray<long> axesChanged;
+  readonly attribute FrozenArray<long> buttonsChanged;
+  readonly attribute FrozenArray<long> buttonsPressed;
+  readonly attribute FrozenArray<long> buttonsReleased;
+};
+
+```
+
+### Dependencies on non-stable features
+
+None identified at this stage. The proposal builds upon the existing Gamepad API.
+
+## Alternatives considered
+
+### Alternative 1: axischange and buttonchange events
+
+axischange event: that fires when a member of the Gamepad's axes array changes.
+buttonchange event: that fires when a GamepadButton's value attribute changes.
+
+- Event Overload: A gamepad with multiple axes and analog buttons may trigger many events within a single input frame, especially if several axis or button values are modified at the same time.
+- Sequential Input Processing: Applications may prefer to process related inputs simultaneously rather than one after the other. For example, both the X and Y axes of a thumbstick are usually handled together.
+
+## Accessibility, Privacy, and Security Considerations
+To prevent abuse and fingerprinting:
+User Gesture Required:  Gamepad events won’t start firing until a user interacts with the gamepad (e.g., pressing a button). [Gamepad user gesture](https://www.w3.org/TR/gamepad/#dfn-gamepad-user-gesture)
+
+Limit Persistent Tracking (fingerprinting): gamepadchange event will not expose device-specific identifiers. By default, no gamepad state is exposed to the tab even when gamepads are connected. [Fingerprinting in Web](https://www.w3.org/TR/fingerprinting-guidance/)
+
+## Stakeholder Feedback / Opposition
+Firefox: [#554 Gamepad API button and axis event](https://github.com/mozilla/standards-positions/issues/554)
+
+WebKit JS: Positive
```

Comment:
What is "WebKit JS"? For these positions, basically what I have observed is that Firefox=Mozilla and Safari=WebKit.